_posts/2020-01-29-#17487.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: pr
+date: 2020-01-29
+title: "Coins: allow Flush() without cache drop"
+pr: 17487
+authors: [jamesob]
+components: ["utxo db and indexes"]
+host: jamesob
+status: upcoming
+---
+
+## Notes
+
+The UTXO cache is a critical datastructure in Bitcoin for both correctness and
+performance. It's responsible for maintaining a view of the spendable coins
+based upon the transactions in blocks. The UTXO cache is just a denormalization
+(i.e. a copy of) of the information found in blocks, but it is often consulted
+exclusively for various things for performance reasons. For example while
+validating the spends in an incoming block, we consult the UTXO cache (instead
+of digging the coins out of the blockchain) because we want this process to be
+as fast as possible.
+
+Because the UTXO set is accessed so frequently, we would like its contents
+available as quickly as possible. This amounts to having as much of the set
+in-memory as possible. The issue is that (at the moment) the in-memory
+representation of the UTXO set is more than 8GB, and obviously not all hosts
+running Bitcoin have that much memory. 
+
+For that reason, the UTXO cache [is stratified across several
+layers](https://jameso.be/dev++2018/#54): some on-disk, and some in-memory. The
+`-dbcache` parameter controls how much memory we allocate to the in-memory
+portion. As we validate blocks, we pull unspent coins that we look up from disk
+into memory until we run out of allocated memory (as indicated by [this
+logic](https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp#L2198-L2220)).
```

Comment:
I suggest you use persistent commit hash links (eg https://github.com/bitcoin/bitcoin/blob/28fbe68fdcac2a06f359b1e48555a3d23015c2b7/src/validation.cpp#L2198-L2220) instead of branch links, which become outdated whenever the branch is updated.